### PR TITLE
feat: add state manager client

### PR DESCRIPTION
**Pull request by dhercher**
Adding a client to manage state (connections and configurations)

The client should transparently know which file system/source to use for the state.
Then have helper commands for any stateful operations we run, currently CRUD for connection management

### CODE REVIEW

File: data_validation/__main__.py [hunk 1]
```diff
@@ -131,27 +134,28 @@ def build_config_managers_from_args(args):
             threshold = args.threshold
         labels = cli_tools.get_labels(args.labels)
 
-    source_client = clients.get_data_client(source_conn)
-    target_client = clients.get_data_client(target_conn)
+    mgr = state_manager.StateManager()
+    source_client = clients.get_data_client(mgr.get_connection_config(args.source_conn))
+    target_client = clients.get_data_client(mgr.get_connection_config(args.source_conn))
```
Comment (nehanene15) [inline data_validation/__main__.py]:
```suggestion
    target_client = clients.get_data_client(mgr.get_connection_config(args.target_conn))
```

File: tests/unit/test_state_manager.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    config = manager.get_connection_config(TEST_CONN_NAME)
+    assert config == TEST_CONN
+
+
+def test_create_and_list_connection(capsys, fs):
+    manager = state_manager.StateManager()
+    manager.create_connection(TEST_CONN_NAME, TEST_CONN)
+
+    connections = manager.list_connections()
+    assert connections == [TEST_CONN_NAME]
```
Comment (nehanene15) [inline tests/unit/test_state_manager.py:40]: Could be useful to add tests here for invalid GCS/filesystem paths.

**Review by nehanene15 — CHANGES_REQUESTED** [review]
Looks good, suggested two small changes + adding documentation about setting the 'PSO_DV_CONFIG_HOME' env var would be great

**Review by nehanene15 — APPROVED** [review]
LGTM

**Comment by bbrewington** [comment]
@nehanene15 minor documentation issue, but was this formatting change intentional?

Below screenshot from commit: https://github.com/GoogleCloudPlatform/professional-services-data-validator/commit/e893ea5f723e0e33350410e64e0d588cc96b36cf

Screen Shot 2021-11-13 at 8 20 25 AM

**Comment by nehanene15** [comment]
> @nehanene15 minor documentation issue, but was this formatting change intentional?

Thanks for catching this. I don't believe so; @dhercher correct me if I am wrong.

**Pull request merged**